### DISCUSSION TITLE
`ContextMenu.PlacementTarget` is always `null`

### DISCUSSION THREAD

**Discussion by duck-dev** [AvaloniaUI/Avalonia#16351] [Q&A]
### Describe the bug

In my entire project, wherever I define a `ContextMenu`, its `PlacementTarget` property is always null. I encountered this issue, as was trying to bind to the `DataContext` of the control, for which the `ContextMenu` is defined (which should be referenced by `PlacementTarget`), for the `Command` property of a `MenuItem`. But I noticed that the `PlacementTarget` is null. Now, I thought this might have something to do that this target is defined as an `ItemTemplate` of an `ItemsControl`, but trying it on a simple `Panel` inside the main window shows the same result.

### To Reproduce

I couldn't find any exceptions, where `PlacementTarget` was actually not `null`, so really any `ContextMenu` defined on any control.
```cs
<Control.ContextMenu>
        <ContextMenu>
                <MenuItem Header="Something"/>
        </ContextMenu>
</Panel.ContextMenu>
```

### Expected behavior

The `PlacementTarget` of a `ContextMenu` should not be `null`. As we're talking about the `ContextMenu` and not a `Flyout`, it is not directly possible to control the opening of a `ContextMenu`, meaning I shouldn't be responsible for setting this property.

### Avalonia version

11.0.11

### OS

Linux

**Comment by maxkatz6** [comment]
> I couldn't find any exceptions, where PlacementTarget was actually not null

It's not null, when you set an actual value to it. Otherwise, it will fallback to the control that requests context menu. 
Like other Placement properties, PlacementTarget is a popup (in this case - context menu) configuration property.

**Reply by duck-dev** [reply]
@maxkatz6 After seeing the source code for `ContextMenu`, I understand what you mean with the fallback value. And of course it makes sense, because it is a get-set property. I suppose you're referring to this: https://github.com/AvaloniaUI/Avalonia/blob/5521baa827752f05338559c0f3730d4c494f4ba1/src/Avalonia.Controls/ContextMenu.cs#L268

But there is no way to retrieve this control in XAML afterwards, because `PlacementTarget` is `null`. How could I go about that to reference the target for the `Command` of a `MenuItem`? Especially if the `ContextMenu` is defined in a template.

**Reply by maxkatz6** [reply]
ContextMenu doesn't seem to have API for that.
Consider using ContextFlyout with MenuFlyout instead.
